policy-engine/src/main.rs

```diff
@@ -59,10 +61,12 @@ fn main() -> Result<(), Error> {
     .start();
 
     // Main service.
+    let plugins = Arc::new(settings.policy_plugins()?);
     let state = AppState {
         mandatory_params: settings.mandatory_client_parameters.clone(),
         upstream: settings.upstream.clone(),
         path_prefix: settings.path_prefix.clone(),
+        plugins: plugins,
```

Comment:
We should verify what happens to the plugins if the AppState is cloned with regard to mutable, persistent fields in the plugins.

Reply:
I've updated this logic to use the syntax recommend in https://doc.rust-lang.org/std/sync/struct.Arc.html#cloning-references.

That one makes it clearer that the code is creating a new reference, rather than duplicating the whole content.